### PR TITLE
return error on args passed to no-args oc cmd

### LINKED ISSUE #4021: Commands that don't accept args should fail when args are passed
Not all commands are properly restricting the use of arguments.  This means that they will silently ignore these args, resulting in unexpected behavior.  For example:

```
oc whoami --context somevalue
```

should fail, but it runs without complaint.

We need to do a check of all the commands to make sure they have validation for this condition.

@stevekuznetsov ptal

### PR DESCRIPTION
**Pull request by juanvallejo**
Returns 
```
error: no arguments should be provided
See 'oc whoami -h' for help and examples.
```

when the `oc whoami` command is run with arguments.

All commands have been checked to make sure they return an error if args are passed when they don't take any. All `origin` commands are fine. There are a few others that do not fail, but are part of the k8s dep

fixes #4021

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Up to @deads2k but a one-liner to test this in `test-cmd` would be easy and might be useful

**Comment by deads2k** [comment]
@juanvallejo scan the code and standardize our string.  

`kcmdutil.CheckErr(kcmdutil.UsageError(cmd, "%s", "no arguments should be provided"))` is the correct failure mode.  It should either be done in `Complete` (if the command has that function) or the `Run` function itself.

**Comment by deads2k** [comment]
In the `Complete` method, you return, in the `Run` you `CheckErr`.

**Comment by juanvallejo** [comment]
@deads2k Okay, I went ahead and updated it (added a test one-liner as well). This file does not seem to follow convention of having a Complete or Run method, however every failure in its `RunWhoAmI` function returns an error which gets passed on to `kcmdutil.CheckError` by the caller function.

https://github.com/juanvallejo/origin/blob/jvallejo_clean-cli-cmds-with-no-args/pkg/cmd/cli/cmd/whoami.go#L62-L71

**Comment by deads2k** [comment]
@juanvallejo find all the other commands that already have this check with a different message and get those too.

File: pkg/cmd/cli/cmd/whoami.go [hunk 1]
```diff
@@ -57,6 +58,9 @@ func (o WhoAmIOptions) WhoAmI() (*userapi.User, error) {
 }
 
 func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, o *WhoAmIOptions) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments should be provided")
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/whoami.go]: @fabianofranz this should be consistent in all commands that don't accept any arguments

File: pkg/cmd/cli/cmd/whoami.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package cmd
 
 import (
+	// "errors"
```
Comment (stevekuznetsov) [inline pkg/cmd/cli/cmd/whoami.go]: @juanvallejo remove this

**Comment by juanvallejo** [comment]
@deads2k @stevekuznetsov Did an `ack` search on `pkg/cmd/cli/cmd` and updated `pkg/cmd/cli/cmd/logout.go` and `pkg/cmd/cli/cmd/status.go` to have consistent no args error output. Added one-line test cases for both of these files as well in `hack/test-cmd.go`

**Comment by stevekuznetsov** [comment]
How are you determining that a command doesn't accept any args?

**Comment by deads2k** [comment]
missed prune deployments, prune images, bootstrap project template, bootstrap policy file, create client, create client cert, create error template, create key pair, create kubeconfig, create login template, and a half dozen others.

**Comment by deads2k** [comment]
> How are you determining that a command doesn't accept any args?

Actually, yeah, what are you searching for?  Almost none of our commands are using the `kcmdutil.UsageError` like they should be.

**Comment by juanvallejo** [comment]
@deads2k @stevekuznetsov 
> How are you determining that a command doesn't accept any args?

I've been doing `oc help`, going through each command, and for the ones located in `pkg/cmd/cli/cmd/`, I've been reading their usage to see if they take any args or not.

With `ack`, I simply did the search for `arguments` inside of that same folder and looked at the ones that were already returning some for of error when arguments were passed.

**Comment by juanvallejo** [comment]
> missed prune deployments, prune images, bootstrap project template, bootstrap policy file, create client, create client cert, create error template, create key pair, create kubeconfig, create login template, and a half dozen others.

@deads2k or @stevekuznetsov any chance you could point me in the right direction with this whenever you have a minute please?

**Comment by stevekuznetsov** [comment]
- prune deployments --> `oadm prune deployments` 
 - prune images --> `oadm prune images`

 Most look like admin commands, which are under  `oadm` and live in `pkg/cmd/admin`

When I was originally looking into this issue I wrote a script to recursively traverse the `openshift --help` tree to find commands that didn't say they accepted any args. I don't have that script anymore, sadly, as I don't think it survived my F21-->F23 migration, but you could look into something similar.

**Comment by deads2k** [comment]
I did the low-tech grep for `len(args) != 0` and `len(args) > 0` inside of the entire `pkg/cmd` tree.

File: hack/test-cmd.sh [hunk 3]
```diff
@@ -269,6 +269,8 @@ os::cmd::expect_failure_and_text "oc login https://server1 https://server2.com"
 os::cmd::expect_success "oc login ${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything --api-version=v1"
 os::cmd::expect_success_and_text "cat ${HOME}/.kube/config" "v1"
 os::cmd::expect_success 'oc logout'
+# fails after receiving extraneous arguments with logout cmd
+os::cmd::expect_failure_and_text "oc logout test_arg" "no arguments"
```
Comment (fabianofranz) [inline hack/test-cmd.sh]: test the entire string

File: hack/test-cmd.sh [hunk 4]
```diff
@@ -304,6 +306,10 @@ os::cmd::expect_success_and_text 'oc whoami' 'test-user'
 os::cmd::expect_success_and_text "oc whoami --config='${MASTER_CONFIG_DIR}/admin.kubeconfig'" 'system:admin'
 os::cmd::expect_success_and_text 'oc whoami -t' '.'
 os::cmd::expect_success_and_text 'oc whoami -c' '.'
+os::cmd::expect_failure_and_text 'oc whoami -c -t test_arg' 'no arguments'
```
Comment (fabianofranz) [inline hack/test-cmd.sh]: same here and the other tests

**Pull request closed without merging**